pkg/application/registry/application/strategy.go

```diff
@@ -236,6 +249,9 @@ func NewFinalizerStrategy(strategy *Strategy) *FinalizeStrategy {
 func (FinalizeStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
 	newApplication := obj.(*application.App)
 	oldApplication := old.(*application.App)
+	finalizers := newApplication.Spec.Finalizers
+	newApplication.Spec = oldApplication.Spec
+	newApplication.Spec.Finalizers = finalizers
```

Comment:
why we need these code? why other resources like cluster doesn't need?

Reply:
for consistency, I remove this